src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java

```diff
@@ -170,11 +172,17 @@
 
         subscriptionNamesList.forEach(subscriptionName -> {
             LOG.debug("Subscription deleting has been started: " + subscriptionName);
-            if (subscriptionService.deleteSubscription(subscriptionName)) {
-                LOG.debug("Subscription is deleted successfully: " + subscriptionName);
-            } else {
-                LOG.error("Subscription to delete is not found: " + subscriptionName);
-                errorMap.put(subscriptionName, SUBSCRIPTION_NOT_FOUND);
+
+            try {
+                if (subscriptionService.deleteSubscription(subscriptionName)) {
+                    LOG.debug("Subscription is deleted successfully: " + subscriptionName);
+                } else {
+                    LOG.error("Subscription to delete is not found: " + subscriptionName);
+                    errorMap.put(subscriptionName, SUBSCRIPTION_NOT_FOUND);
+                }
+            } catch (NotOwnerException e) {
+                LOG.error("Unauthorized, wrong username: " + subscriptionName);
```

Comment:
you log two error messages for the same exception. I thin it is better to remove the error logging here and keep the error logging where the exception is created.

Alternatively create an exception with the message you log when you create it and log it here.

Reply:
Done.